immunomind/immunarch
# Immunarch 0.6.8 release

- tutorials update
- reverted unneeded changes
- repFilter tutorial
- v1_introduction reverted
- Update v2_data.Rmd
- Update v2_data.Rmd
- Create v7_fixvis.html
- Revert "Create v7_fixvis.html"
- Update vignettes/v2_data.Rmd
- fixed fixvis
- Update vignettes/v2_data.Rmd
- Update vignettes/v2_data.Rmd
- Update vignettes/v2_data.Rmd
- Update vignettes/v2_data.Rmd
- Update vignettes/v2_data.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update repFilter_v3.Rmd
- Update repFilter_v3.Rmd
- Update v2_data.Rmd
- draft of repDist function fo distance calculations
- new changes with scdata
- Revert "Update v2_data.Rmd"
- Revert "Revert "Update v2_data.Rmd""
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update vignettes/web_only/repFilter_v3.Rmd
- Update repFilter_v3.Rmd
- Update .gitignore
- Update _pkgdown.yml
- Update _pkgdown.yml
- update links
- Update distance.R
- documentation update
- documentation update
- Update _pkgdown.yml
- Update R/distance.R
- Update R/distance.R
- tests added
- Update R/distance.R
- Update R/distance.R
- Update distance.R
- Negative and positive test fixed
- Naming changes
- Update R/distance.R
- Update R/distance.R
- Update R/distance.R
- Update R/distance.R
- Update R/distance.R
- Update R/distance.R
- Update tests/testthat/test-distance.R
- Update R/distance.R
- Update R/distance.R
- Update R/distance.R
- Update seqDist.Rd
- version bump
- UPD: removed typo in the SeqDist description
- fix warnings; prepare necessary columns
- introduced .dataset_type argument for repLoad
- reverted some unwanted changes
- improved fix.alleles function
- updates from new version of styler
- more fix.allele optimizations
- first implementation of BCR columns in MiXCR data
- removed generated file after updating R packages
- BCR column presence check utility function
- support for FRx nucleotide and amino acid sequences
- fixes for tests and CRAN checks
- CDR3 begin/end parsing from refPoints
- RcppExports.R returned to repository
- updata tutorials
- grouping by variables(genes as default) and seq length added
- Update seqDist.Rd
- Update R/distance.R
- Update tests/testthat/test-distance.R
- Update R/distance.R
- minor updates
- seqCluster function + typos fixing in seqDist
- styling files
- Update R/seqCluster.R
- Update R/seqCluster.R
- Update R/seqCluster.R
- hotfix for MiXCR parser: don't crash when extra CDR/FR columns are not present
- Update test-seqCluster.R
- Update tests/testthat/test-seqCluster.R
- added parameter for vis_heatmap: .axis.text.size
- fixes for docs
- support for C gene alignments in MiXCR format; bugfix for "\\|" in D genes
- Fix missing D alignment when loading a single clone
- Fix missing D alignment when loading a single clone
- Fix missing D alignment when loading a single clone
- seqGermline prototype; for now with hardcoded reference
- load reference sequences from VDJgermlines library
- repGermline fixes and docs
- MiXCR parser: fix bug with "dalignments" column
- MiXCR parser: don't remove alleles
- added alleles stripping to geneUsage
- better handling of mixing immdata and immdata$data
- repGermline fixes
- better return part
- error message fix
- docs update
- MiXCR parser: fix bug with "dalignments" column
- align lineage docs; code WIP
- Dockerfile update
- Dockerfile: added muscle and phylip which are required for BCR pipeline
- germline hotfixes: species; removed VDJgermlines dependency
- added ape package to dependencies
- Dockerfile: added muscle and phylip which are required for BCR pipeline
- germline hotfixes: species; removed VDJgermlines dependency
- check docker image for Immunarch existence
- check for missing alleles in repGermline; warning message
- updated gemline calculation algorithm
- germline: pass "V" instead of "IGHV"
- repAlignLineage implementation
- germline: pass "V" instead of "IGHV"
- code style fixes
- code style fixes
- added warning about unexpected NAs
- style fixes
- style fix
- Update R/align_lineage.R
- Revert "Update R/align_lineage.R"
- changed `%>% return()` to `return(.data)` everywhere
- Update R/align_lineage.R
- fixed typo in docs
- repGermline: better data cleanup and validation
- comment fix
- replace *00 to *01 in alleles for germline calculation
- added BCR data for examples
- Germline algorithms update
- drop missing V.end and J.start instead of autodetecting
- additional check for empty dataframe
- updated docs and arguments for the planned changes
- repAlignLineage updates WIP
- validate_chains_length() min length parameter update
- repGermline: drop sequences with too short V/J chains
- bcrdata changed to new data (better, filtered)
- bcrdata changed to new data (better, filtered)
- Updated docs related to germline and bcrdata
- bugfix: removed unimplemented function from NAMESPACE
- Germline: min_nuc_outside_cdr3 made public parameter
- align lineage: update docs, group by both clusters and germlines; use MUSCLE
- Update for seqDist: use V.first/J.first by default; some frequent operations made generic in tools.R
- * seqDist: use V.first/J.first by default * repGermline: min_nuc_outside_cdr3 parameter is now public * some frequent operations made generic in tools.R
- for_sample_or_list renamed to apply_to_sample_or_list
- alignLineage WIP
- fix seqCluster cluster name assignment
- changes to namespace
- alignLineage: implementation of new features
- repAlignLineage: various bugfixes and improvements; first working version
- repClonalFamily WIP
- phylip: various bugfixes; working multi-threading implementation
- small fixes
- MiXCR parser update: 3'V and 3'J deletions/insertions loading into table
- ref_only_first parameter in Germline to prevent dataset growth; avoid putting reference in global environment
- formula fixes for germline V/J trimming
- bcrdata updated with new fields supported by MiXCR parser
- added non-verbose output for repAlignLineage; TRUE by default
- repAlignLineage docs update
- repClonalLineage WIP
- Check for file existence in repLoad()
- fix in docs
- Phylip: implemented outfile parsers
- Added trunk length calculation
- Fixes for CRAN release
- Fixes: docs, CRAN and IDE warnings
- CRAN fixes
- CRAN fixes
- CRAN fixes; bcrdata trimmed to 1000 clonotypes; better compression for datasets
